### ISSUE TITLE
Bg-utils converting json config files to yaml

### ISSUE THREAD

**Issue by hazmat345** [beer-garden/beer-garden#166]
Running application with json config files that had been migrated from 2.1.x. Running `migrate_*_config` correctly kept the files json, but running the application converted them to yaml.

**Comment by hazmat345** [comment]
@loganasherjones take a look at https://github.com/beer-garden/bg-utils/commit/46c24683728e526e74b771a8ca6f0d747b6a6a0b when you can.

We had a bug with how we were doing the comparison to determine if a configuration file was 'new'. We were loading the Box object from both files and then comparing the Boxes. But the structure of the Box object is much more dependent on the spec, and since we were using the same spec for both in most cases the resulting box object would be the same. In fact, it probably would have indicated an error on our part in the actual spec if they weren't.

We were doing an additional json / yaml check previously (files were always 'different' if the original was json since we were forcing yaml) that was causing some of the tests to 'pass'.

I ran into this same issue (how to tell if the config files are equivalent) when I was making the configuration migration integration tests. I just parsed the file according to the type and then compared the data structures. That seems to be good enough, so I did that here as well.

Now the tests pass (correctly, I think). I also see correct behavior when running the apps for real. It actually 'fixed' the brew-view dev_conf/config.yml by alphabetizing some items. I ran bartender with the old flat-style json and it correctly converted it to the new spec, kept it as json, and stored the old file as the datestamped json file. In both cases restarting the applications *did not* result in additional 'new' temp file being generated (I know we saw that at one point).

Bottom line - I think it's doing what we want. When I updated the dev server I manually called `migrate_*_config`, but I'm pretty confident at this point that I didn't need to and the application would have handled it.

I want that LGTM since these changes were more extensive than I was expecting :smile:

**Comment by loganasherjones** [comment]
👍
